OpenRA.Game/Map/Map.cs

```diff
@@ -651,6 +651,39 @@ public byte[] SavePreview()
 			var resources = Rules.Actors["world"].TraitInfos<ResourceTypeInfo>()
 				.ToDictionary(r => r.ResourceType, r => r.TerrainType);
 
+			var actorTypes = Rules.Actors.Values.Where(a => a.HasTraitInfo<RenderOnMapPreviewInfo>());
```

Comment:
Can you please implement this via an `IMapPreviewSignature` interface that mirrors `IRadarSignature`, and then use this on `AppearsOnRadar` (IMO this should also be renamed to `AppearsOnMapPreview` for consistency with the radar case) and `ResourceLayer`?

`RenderOnMapPreview` and `ResourceType` can then move to Mods.Common.
Removing another hardcoded engine-level resource layer reference should make @IceReaper happy.